src/client/content_cao.cpp

```diff
@@ -325,7 +325,9 @@ void TestCAO::processMessage(const std::string &data)
 #include "clientobject.h"
 
 GenericCAO::GenericCAO(Client *client, ClientEnvironment *env):
-		ClientActiveObject(0, client, env)
+		ClientActiveObject(0, client, env),
+		m_material_type(video::EMT_TRANSPARENT_ALPHA_CHANNEL_REF),
+		m_material_type_param(0.5f)
```

Comment:
are these values actually used? otherwise a bit confusing to init them here IMO

Reply:
`m_material_type` happens to be assigned in every case. I felt like initializing the variables to a sane default that would then be changed later was a more reliable solution compared to potentially end up with uninitialized values.

Reply:
If they're always meant to be overwritten I suggest putting clearly invalid values in there so this mistake is noticed.